dev/tasks/exec/tasks/exec.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const tools = require( '../../../lib/tools' );
+const git = require( '../../../lib/git' );
+const path = require( 'path' );
+const log = require( '../../../lib/log' );
+const merge = require( 'merge-stream' );
+
+/**
+ * @param {Function} execTask Task to use on each dependency.
+ * @param {String} ckeditor5Path Path to main CKEditor5 repository.
+ * @param {Object} packageJSON Parsed package.json file from CKEditor5 repository.
+ * @param {String} workspaceRoot Relative path to workspace root.
+ * @param {Boolean} dryRun
+ */
+module.exports = ( execTask, ckeditor5Path, packageJSON, workspaceRoot, params ) => {
+	const workspaceAbsolutePath = path.join( ckeditor5Path, workspaceRoot );
+
+	// Get all CKEditor dependencies from package.json.
+	const dependencies = tools.getCKEditorDependencies( packageJSON.dependencies );
+	const streams = merge();
+
+	if ( dependencies ) {
```

Comment:
But that doesn't explain why we need to check all that. Can't we just take CKEditor directories? Perhaps I just don't understand what the existing functions do but that may mean that they should be improved. The logic you put here, with getting deps and then dirs should be enclosed in some other, reusable function.